core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -244,6 +246,9 @@ public void commit() {
               updated = base.replaceCurrentSnapshot(newSnapshot);
             }
 
+            // reset sequence number to null so that the sequence number can be updated when retry
+            this.sequenceNumber = null;
```

Comment:
Because the sequence number cannot be unique, so it has to reset to null for next use.